bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/AudioFormat.java

```diff
@@ -304,26 +305,28 @@ public boolean isCompatible(@Nullable AudioFormat audioFormat) {
     }
 
     /**
-     * Determines the best match between a list of audio formats supported by a source and a sink.
+     * Determines the best match between two list of audio formats.
+     * Used to check for compatibility between audio sources and sinks
+     * and with services that depend on them.
      *
-     * @param inputs the supported audio formats of an audio source
-     * @param outputs the supported audio formats of an audio sink
-     * @return the best matching format or null, if source and sink are incompatible
+     * @param requiredFormats audio formats supported by first source/sink/service
+     * @param availableFormats audio formats supported by second source/sink/service
+     * @return the best matching format or null, if all are incompatible
      */
-    public static @Nullable AudioFormat getBestMatch(Set<AudioFormat> inputs, Set<AudioFormat> outputs) {
-        AudioFormat preferredFormat = getPreferredFormat(inputs);
-        if (preferredFormat != null) {
-            for (AudioFormat output : outputs) {
-                if (output.isCompatible(preferredFormat)) {
-                    return preferredFormat;
-                } else {
-                    for (AudioFormat input : inputs) {
-                        if (output.isCompatible(input)) {
-                            return input;
-                        }
-                    }
-                }
+    public static @Nullable AudioFormat getBestMatch(Set<AudioFormat> requiredFormats,
```

Comment:
My feeling is that you potentially changed the result of getBestMatch and I don't know if this is a good change or not.

Reply:
I think that's the desired behavior as the previous one did not ensure the "best" (the one selected by getPreferredFormat from the list of formats) was returned. 
I can move this to a separate PR if you think is better.

Reply:
Sorry, you were right.
I was not checking all available formats as the previous implementation did.
Now is fixed.
This implementation should work like the previous one but always prioritizing by getPreferredFormat.

Reply:
You are right, I will remove the changes to this method